### ISSUE TITLE
CSS and JS caching is defeated by URL parameters

### ISSUE THREAD

**Issue by meyergru** [skeeks-semenov/yii2-assets-auto-compress#28]
I found that because the compressed resources are suffixed by a '?v=' parameter that has the file creation time, effectively resource caching is defeated.

Why is this? The name of the file is already made of the MD5 of the file content, so why is there another parameter which is not really used anyway? I assume you tried to defeat the resource caching if the file was modified, but in that case, the MD5 would differ anyway. By adding an URL parameter, nothing gets cached at all.

If I am correct, it would suffice to just remove the code lines where the '?v=' is added.

**Comment by skeeks-semenov** [comment]
Yes, this is done to clear the cache. And it need not be removed.

The logic is that under any scenario, checked md5 just from a set of files. If a single file is updated, the yii2 change it ?v=value and then md5 of the data set in the same change. This is easy to do and does not require the reading of all the files and comparing them md5 values.

https://github.com/skeeks-semenov/yii2-assets-auto-compress/blob/master/AssetsAutoCompressComponent.php#L353

**Comment by meyergru** [comment]
I think I understand how it works, but the function you are pointing to does:

a.) create an MD5 for the resulting .js file name from the source files.
b.) take the filemtime() of the created .js file containing all the single .js files.

This is redundant and does nothing more than just a.) because the url parameter '?v=' is not used except for file cache invalidating in case one of the single .js files changes. But in that case, both the MD5 file name and the filemtime() change.

The problem is that for resources with URL parameters, there will be no caching AT ALL, regardless of the parameter's value. At least that is what I saw when I used Firefox inspector - the resulting .js is always transferred on reload.
